modules/local/format_prodigal.nf

```diff
@@ -20,9 +20,13 @@ process FORMAT_PRODIGAL {
     script:
     def args = task.ext.args   ?: ''
     prefix   = task.ext.prefix ?: "${meta.id}"
+    input    = gff =~ /\.gz$/ ? gff.name.take(gff.name.lastIndexOf('.')) : gff
+    gunzip   = gff =~ /\.gz$/ ? "gunzip -c ${gff} > ${input}" : ""
 
     """
-    sed 's/\\(\\(k[0-9]\\+_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' $gff > ${prefix}_format.gff
+    $gunzip
+
+    sed 's/\\(\\(k[0-9]\\+_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' $input > ${prefix}_format.gff
     sed -i 's/\\(\\(NODE_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' ${prefix}_format.gff
```

Comment:
Why are you doing this here, better build a full pipeline above. (Remember to use `\\` to place commands on multiple lines.)